docs/concepts/models.md

```diff
@@ -168,6 +172,51 @@ Models possess the following methods and attributes:
     See [Changes to `pydantic.BaseModel`](../migration.md#changes-to-pydanticbasemodel) in the
     [Migration Guide](../migration.md) for details on changes from Pydantic V1.
 
+## Data conversion
+
+Pydantic may cast input data to force it to conform to model field types,
+and in some cases this may result in a loss of information.
+For example:
+
+```python
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    a: int
+    b: float
+    c: str
+
+
+print(Model(a=3.000, b='2.72', c=b'binary data').model_dump())
+#> {'a': 3, 'b': 2.72, 'c': 'binary data'}
+```
+
+This is a deliberate decision of Pydantic, and is frequently the most useful approach. See
+[here](https://github.com/pydantic/pydantic/issues/578) for a longer discussion on the subject.
+
+This is also the case for collections. In most cases, you shouldn't make use of abstract container classes
+and just use a concrete type, such as [`list`][]:
```

Comment:
This is added by me, as I think a common misconception is to use abstract containers as types thinking this will allow (in the case of `collections.abc.Sequence`) both lists and tuples to validate, while in fact this is already supported by Pydantic.

Reply:
I like this example overall.

I think we should:
* Link to the related [performance section](https://docs.pydantic.dev/latest/concepts/performance/#sequence-vs-list-or-tuple-mapping-vs-dict)
* Give a bit more context re efficiency here - the more specific you can be with a type, the better (sort of in a philosophical sense)

Reply:
I added the link. Regarding the second bullet point, not sure what you mean. Do you have a suggestion?

Reply:
Sure - basically, in one line or two, you can explain that generally, the more specific a type, the faster validation will be, as we don't have to perform as many checks / coercion attempts.